packages/@pollyjs/adapter-fetch/src/index.js

```diff
@@ -26,23 +28,72 @@ export default class FetchAdapter extends Adapter {
       );
     }
 
-    ['fetch', 'Response', 'Headers'].forEach(key =>
+    ['fetch', 'Request', 'Response', 'Headers'].forEach(key =>
       this.assert(`${key} global not found.`, !!(context && context[key]))
     );
     this.assert(
       'Running concurrent fetch adapters is unsupported, stop any running Polly instances.',
-      !context.fetch[IS_STUBBED]
+      !context.fetch[IS_STUBBED] && !context.Request[IS_STUBBED]
     );
 
-    this.native = context.fetch;
+    this.nativeFetch = context.fetch;
+    this.NativeRequest = context.Request;
+
+    /*
+      Patch the Request class so we can store all the passed in options. This
+      allows us the access the `body` directly instead of having to do
+      `await req.blob()` as well as not having to hard code each option we want
+      to extract from the Request instance.
+    */
+    context.Request = class Request extends context.Request {
+      constructor(url, options) {
+        super(url, options);
+
+        let args;
+
+        options = options || {};
+
+        /*
+          The Request constructor can receive another Request instance as
+          the first argument so we use its arguments and merge it with the
+          new options.
+        */
+        if (typeof url === 'object' && url[IS_REQUEST]) {
+          const reqArgs = url[REQUEST_ARGUMENTS];
+
+          args = { ...reqArgs, options: { ...reqArgs.options, ...options } };
+        } else {
+          args = { url, options };
+        }
+
+        defineProperty(this, IS_REQUEST, { value: true });
+        defineProperty(this, REQUEST_ARGUMENTS, { value: args });
+      }
+
+      clone() {
+        return new Request(this);
+      }
+    };
+
+    defineProperty(context.Request, IS_STUBBED, { value: true });
 
     context.fetch = (url, options = {}) => {
+      let respond;
+
       // Support Request object
-      if (typeof url === 'object' && 'url' in url) {
-        url = url.url;
+      if (typeof url === 'object' && url[IS_REQUEST]) {
```

Comment:
Could use `instanceof Request` here instead and would eliminate the additional `IS_REQUEST` prop

Reply:
That would break if an actual native request gets passed in since we extend the native Request class.

```js
class R extends window.Request {}

new R('/') instanceof window.Request // true
new R('/') instanceof R // true
```

Reply:
Can compare `instanceOfR.constructor === R`

Reply:
Actually I'm not sure I understand your comment.

Are you saying you worry that an instance of `Request,` that isn't an instance of our `ExtendedRequest` class and instead the native `Request`, would get through an `instanceof ExtendedRequest` check?  That shouldn't be the case if so.

Reply:
Ah, yeah you're totally right. My mistake above is not always comparing to the ExtendedRequest 🤦‍♂. I'll make the change to use `instanceof` 😄